Prepare repair for task manager integration

Deexie opened the pull request:
The PR prepares repair for task manager integration:
- Creates repair_module
- Keeps repair_module in repair_service
- Moves tracker methods to repair_module
- Changes UUID to task_id in repair module

Deexie commented on api/api-doc/task_manager.json:
```diff
@@ -49,6 +49,14 @@
                         "type":"string",
                         "paramType":"path"
                     },
+                    {
```
#11821

Deexie commented on tasks/task_manager.hh:
```diff
@@ -14,13 +14,13 @@
 #include <seastar/coroutine/parallel_for_each.hh>
```
#11821

Deexie commented on api/task_manager_test.cc:
```diff
@@ -52,7 +52,7 @@ void set_task_manager_test(http_context& ctx, routes& r, db::config& cfg) {
         it = req->query_parameters.find("entity");
```
#11821

Deexie commented on tasks/task_manager.hh:
```diff
@@ -318,37 +318,37 @@ public:
             co_await _gate.close();
```
#11821

scylladb-promoter commented:
CI state **`SUCCESS`** - https://jenkins.scylladb.com/job/releng/job/Scylla-CI/2792/

bhalevy reviewed (commented):
LGTM overall.
Some mnor comments and room for further cleanups as follow up when moving more functionality to the generic task manager classes.

denesb commented on repair/repair_task.hh:
```diff
@@ -0,0 +1,33 @@
+
+class repair_task_impl : public tasks::task_manager::task::impl {
+public:
+    repair_task_impl(tasks::task_manager::module_ptr module, tasks::task_id id, unsigned sequence_number = 0, std::string keyspace = "", std::string table = "", std::string type = "", std::string entity = "", tasks::task_id parent_id = tasks::task_id::create_null_id())
```
Why so many default parameters. I noticed this being a repeating pattern in task related code. Is there some user of this API which is significantly helped by all this?

denesb commented on repair/repair.hh:
```diff
@@ -64,7 +64,15 @@ struct repair_uniq_id {
     int id;
     // A UUID to identifiy a repair job. We will transit to use UUID over the integer ID.
-    tasks::task_id uuid;
+    tasks::task_info task_data;
```
The comment above needs an update.

denesb commented on repair/repair.cc:
```diff
@@ -1002,7 +1001,7 @@ int repair_service::do_repair_start(sstring keyspace, std::unordered_map<sstring
     // nothing to repair, and return 0. "nodetool repair" prints in this case
     // that "Nothing to repair for keyspace '...'". We don't have such a case
     // yet. Real ids returned by next_repair_command() will be >= 1.
-    auto id = repair_tracker().next_repair_command();
```
Comment above needs update.

denesb reviewed (commented):
LGTM, some small comments.

Deexie commented:
- rebased on next
- deleted default parameters from `repair_task_impl`
- updated comments concerning `repair_uniq_id`
- `task_data` -> `task_info` in `repair_uniq_id`

scylladb-promoter commented:
CI state **`SUCCESS`** - https://jenkins.scylladb.com/job/releng/job/Scylla-CI/2802/

denesb commented:
@Deexie int he future, please don't set the target branch to `next`, leave it as `master`.
